qt/scientific_interfaces/Indirect/FunctionTemplateBrowser.h

```diff
@@ -48,7 +50,8 @@ class MANTIDQT_INDIRECT_DLL FunctionTemplateBrowser : public QWidget {
   virtual IFunction_sptr getFunction() const = 0;
   virtual void setNumberOfDatasets(int) = 0;
   virtual int getNumberOfDatasets() const = 0;
-  virtual void setDatasetNames(const QStringList &names) = 0;
+  virtual void
+  setDatasets(const QList<QPair<QString, QList<std::size_t>>> &datasets) = 0;
```

Comment:
The nested types are a little difficult to parse and have to be duplicated in quite a few places. At minimum I think a few `using` statements would help to give some types better names:

```
using Spectra = QList<std::size_t>;  // should this be using the spectrum types defined? 
using Dataset = QPair<QString, Spectra>;
```
then `QList<Dataset>` can be used in the function definition or even.

At this level I would also consider whether there is a class or two we are missing rather than using fundamental types?

Reply:
I had the same thought, and I think we are missing some fundamental types.

I am aware that Indirect has a `Spectra` class but I find it to be overly complex for what it should be doing. I think I will define a `Spectra` class and `Dataset` class like you suggested, unless you know of any existing classes which already exist which just need to be incorporated?

Reply:
I really picked that name as a guess that it was a collection of spectrum ids? How does the current Spectra class work - we should be careful about duplicating things.

Reply:
Yes its just a list of spectrum numbers which we want to fit from a workspace.

The indirect `Spectra` class seems to do quite a lot, but might be useful I guess .https://github.com/mantidproject/mantid/blob/f8512fdfa6b66a54fdd9bb3dd1ec7f16c3ff82d8/qt/scientific_interfaces/Indirect/IndirectFitData.h#L32 
 What do you think?

Reply:
@martyngigg The indirect `Spectra` class works fine it seems. I have moved this class into its own file and placed it in the same directory as the `FunctionModel`, and I did the same with `IndexTypes`. I did this because they are general classes that could be useful outside of indirect. Let me know if you can think of a better location for these files